pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -2098,9 +2098,6 @@ public long getLong(int columnIndex) throws SQLException {
     if (isBinary(columnIndex)) {
       int col = columnIndex - 1;
       int oid = fields[col].getOID();
-      if (oid == Oid.INT8) {
-        return ByteConverter.int8(this_row[col], 0);
```

Comment:
The check might be there for performance reasons.
I don't personally think it buys much, however, I'm not a big fan of making a chane for the sake of change.